src/libstore/daemon.cc

```diff
@@ -1037,7 +1037,12 @@ void processConnection(
     if (clientVersion < 0x10a)
         throw Error("the Nix client version is too old");
 
-    auto tunnelLogger = new TunnelLogger(to, clientVersion);
+    WorkerProto::BasicServerConnection conn;
+    conn.to = std::move(to);
+    conn.from = std::move(from);
+    conn.protoVersion = clientVersion;
```

Comment:
Can this use designated initializer syntax? I think so 

```suggestion
    WorkerProto::BasicServerConnection conn {{
        .to = to,
        .from = from,
        .clientVersion = clientVersion,
    }};
```

Reply:
No, that doesn't work with inheritance.